### PR TITLE
Generate Student EDA

### PR DESCRIPTION
**Pull request by AlejandroAdrian**
Signed-off-by: Alejandro <alejandro.adrian@exponentiateam.com>

### CODE REVIEW

File: app/views/students/_student.html.erb [hunk 1]
```diff
@@ -1,10 +1,10 @@
-<div class="shadow-md px-2 mb-3 sm:flex items-stretch border <%= student.deactivated? ?  "border-yellow-300" : "border-gray-100" %>"> 
+<div class="shadow-md px-2 mb-3 sm:flex items-stretch border <%= student.deactivated? ?  "border-yellow-300" : "border-gray-100" %>">
   <div class="border-b border-gray-100 py-1 sm:w-2/5 sm:border-b-0 sm:px-1 md:flex md:flex-col md:justify-center">
     <p class="text-lg font-semibold tracking-tight"><%= student.full_name %></p>
-    <p class="text-sm text-gray-500 capitalize"><%= t("student.yearsold", student_age: student.age) %></p>
+    <p class="text-sm text-gray-500 capitalize"><%= t("student.yearsold", student_age: Date.today.year-student.birthday.year) %></p>
```
Comment (mapreal19) [inline app/views/students/_student.html.erb:4]: extract a method for the student to calculate its age.

File: config/locales/models/student_edas/en.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      mode_of_study: "Mode of study"
+      mode_of_delivery: "Mode of delivery"
+      language: "Language"
+      email_address: "E-mail address"
+      certification_date: "Certification date"
+      course_unit_type: "Course unit type"
+      date: "Date"
+      ECTS_grading_scale_type: "ECTS grading scale type"
+      national_framework_qualifications: "National framework qualifications"
+      percert: "Percent"
```
Comment (mapreal19) [inline config/locales/models/student_edas/en.yml:18]: percent?

File: config/locales/models/student_edas/es.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      mode_of_study: "Modo de estudio"
+      mode_of_delivery: "Modalidad"
+      language: "Idioma"
+      email_address: "Dirección e-mail"
+      certification_date: "Fecha de certificación"
+      course_unit_type: "Tipo de unidad del curso"
+      date: "Fecha"
+      ECTS_grading_scale_type: "Tipo de escala de calificaciones ECTS"
+      national_framework_qualifications: "Marco nacional de calificaciones"
+      percert: "Nota"
```
Comment (mapreal19) [inline config/locales/models/student_edas/es.yml:18]: percent?

File: config/locales/views/student_eda/es.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+es:
+  student_eda:
+    student_eda: "EDA del estudiante"
+    manage: "Aquí puede rellenar los campos del EDA y generarlo"
+    edit: "Aquí puede editar los campos del EDA y re-generarlo"
+    generate: "Generar Expediente Digial del Estudiante"
```
Comment (mapreal19) [inline config/locales/views/student_eda/es.yml:6]: typo digial

File: db/seeds.rb [hunk 5]
```diff
@@ -42,37 +42,37 @@
         {
           name: "Bart",
           first_surname: "Simpson",
-          age: 10
+          birthday: DateTime.new(2011, 1, 1)
```
Comment (mapreal19) [inline db/seeds.rb:45]: you can do `10.years.from_now` too since they don't age ;)

File: spec/requests/student_edas_request_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      student = create(:student)
+
+      get "/students/#{student.id}/student_edas/new"
+
+      expect(response).to be_successful
+    end
+  end
+
+  describe "POST /create" do
+    it "creates student EDA", :vcr do
```
Comment (mapreal19) [inline spec/requests/student_edas_request_spec.rb:20]: vcr is needed?

File: spec/requests/student_edas_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      student = create(:student)
+
+      expect do
+        post "/students/#{student.id}/student_edas",
+             params: { student_eda: attributes_for(:student_eda) }
+      end.to change(StudentEda, :count).by(1)
```
Comment (mapreal19) [inline spec/requests/student_edas_request_spec.rb:26]: don't need to test this change if you're testing the eda is present below

**Review by mapreal19 — CHANGES_REQUESTED** [review]

File: .rubocop.yml [hunk 8]
```diff
@@ -20,6 +20,9 @@ AllCops:
     - 'node_modules/**/*'
     - 'vendor/**/*'
 
+Layout/EndOfLine:
+  Enabled: False
```
[the remark this answers is not in the document]
Comment (AlejandroAdrian) [reply .rubocop.yml:24]: I removed it

File: app/controllers/student_edas_controller.rb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  def find_student
+    Student.find(params[:student_id])
+  end
+
+  def find_eda
+    StudentEda.find(params[:id])
+  end
+
+  def eda_params
+    params.require(:student_eda).permit(:student_code, :mode_of_study, :mode_of_delivery, :language,
```
[the remark this answers is not in the document]
Comment (AlejandroAdrian) [reply app/controllers/student_edas_controller.rb:28]: Done

File: app/models/student_eda.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class StudentEda < ApplicationRecord
+  belongs_to :student
+
+  validate :student_code
```
[the remark this answers is not in the document]
Comment (AlejandroAdrian) [reply app/models/student_eda.rb:4]: its presence is supposed to be validated, I forgot it.

File: app/views/students/_student.html.erb (hunk 1 above)
Comment (AlejandroAdrian) [reply app/views/students/_student.html.erb:4]: Done

File: config/locales/models/student_edas/en.yml (hunk 2 above)
Comment (AlejandroAdrian) [reply config/locales/models/student_edas/en.yml:18]: This is what it says in the standard

File: config/locales/models/student_edas/es.yml (hunk 3 above)
Comment (AlejandroAdrian) [reply config/locales/models/student_edas/es.yml:18]: This is what it says in the standard. I assume it is the mark, but we have no more information

File: config/locales/views/student_eda/es.yml (hunk 4 above)
Comment (AlejandroAdrian) [reply config/locales/views/student_eda/es.yml:6]: Done

File: db/seeds.rb (hunk 5 above)
Comment (AlejandroAdrian) [reply db/seeds.rb:45]: Done

File: spec/requests/student_edas_request_spec.rb (hunk 7 above)
Comment (AlejandroAdrian) [reply spec/requests/student_edas_request_spec.rb:26]: Done

File: spec/requests/student_edas_request_spec.rb (hunk 6 above)
Comment (AlejandroAdrian) [reply spec/requests/student_edas_request_spec.rb:20]: I assumed that a xml file is created with it, but not

File: app/models/student.rb [hunk 11]
```diff
@@ -17,4 +18,8 @@ def full_name
   def deactivated?
     deactivated
   end
+
+  def self.calculate_age(student)
```
Comment (mapreal19) [inline app/models/student.rb:22]: it should be an instance method

```
def age # call it with student.age
  Time.zone.today - birthday.year
end
```

File: spec/requests/student_edas_request_spec.rb (hunk 6 above)
Comment (mapreal19) [reply spec/requests/student_edas_request_spec.rb]: vcr is used when interacting with some API outside of our control

File: app/models/student.rb (hunk 11 above)
Comment (AlejandroAdrian) [reply app/models/student.rb]: Done

File: spec/requests/student_edas_request_spec.rb (hunk 6 above)
Comment (AlejandroAdrian) [reply spec/requests/student_edas_request_spec.rb]: OK, I understand it now

**Review by mapreal19 — APPROVED** [review]

**Pull request merged**